### PR TITLE
PYTHON-2256 Document that a ClientSession cannot be used for multiple operations concurrently

### PR DESCRIPTION
**Pull request by ShaneHarvey**
Jira: https://jira.mongodb.org/browse/PYTHON-2256

### CODE REVIEW

File: pymongo/client_session.py [hunk 1]
```diff
@@ -461,6 +471,10 @@ def callback(session, custom_arg, custom_kwarg=None):
         however, ``with_transaction`` will return without taking further
         action.
 
+        This :class:`ClientSession` is **not thread-safe or fork-safe**. The
+        ``callback`` must not attempt to execute multiple operations
+        concurrently.
```
Comment (prashantmital) [inline pymongo/client_session.py:476]: Suggest rephrasing to something like:

:class:`ClientSession` instances are **not thread-safe or fork-safe**.  Consequently, the ``callback`` must not attempt to use the session for executing multiple concurrent operations

File: pymongo/client_session.py [hunk 2]
```diff
@@ -336,7 +336,17 @@ def _within_time_limit(start_time):
...
 class ClientSession(object):
-    """A session for ordering sequential operations."""
+    """A session for ordering sequential operations.
+
+    :class:`ClientSession` instances are **not thread-safe or fork-safe**.
+    They can only be used by one thread or process at a time. A single
+    :class:`ClientSession` cannot be used to run multiple operations
+    concurrently.
+
+    Should not be called directly by application developers - to create a
```
[the remark this answers is not in the document]
Comment (ShaneHarvey) [reply pymongo/client_session.py]: Done.

File: pymongo/client_session.py (hunk 1 above)
Comment (ShaneHarvey) [reply pymongo/client_session.py]: Done.

**Comment by ShaneHarvey** [comment]
> I am guessing failures are unrelated 

Yep, I filed: https://jira.mongodb.org/browse/PYTHON-2311

**Pull request merged**